lib/c-ares-1.19.0/m4/ax_cxx_compile_stdcxx_11.m4

```diff
@@ -37,6 +37,9 @@
 #serial 9
 
 m4_define([_AX_CXX_COMPILE_STDCXX_11_testbody], [[
+
+#define static_assert _Static_assert
```

Comment:
This change doesn't seem to match neither the  merged c-ares [PR](https://github.com/bradh352/c-ares/blob/941987aa9a314bb552ee6f162fb5536c100af50f/m4/ax_cxx_compile_stdcxx_11.m4) nor the [current](https://github.com/c-ares/c-ares/blob/main/m4/ax_cxx_compile_stdcxx_11.m4) version of the file which means if we update c-ares your change will be lost and that will be considered a regression.

Did I misunderstand anything?

Reply:
There was an upstream fix: https://github.com/bradh352/c-ares/commit/941987aa9a314bb552ee6f162fb5536c100af50f.
Just not in the same way. Since c-ares doesn't have a official release (yet) with that included, I decided to just include a reduced version here. So in other words, there won't be regression if c-ares got synced.

Reply:
Actually, I just saw that what you did was patch our outdated file so I guess that's acceptable if we don't want to update c-ares (which we should tbh)

Reply:
Yes, you should update c-ares over time. And indeed, I patched the outdated file, the new file does the handling somewhat different, but for the exact details, I also have to dive deeper.